test/sbs_test.go

```diff
@@ -209,3 +213,288 @@ func TestSBSNewlyResolvedWritersPublic(t *testing.T) {
+			mkfile("charlie1.txt", "hello alice & bob"),
+		),
+		as(alice,
+			read("alice.txt", "hello bob & charlie"),
+			read("bob1.txt", "hello alice & charlie"),
+			read("bob2.txt", "hello alice & charlie"),
+			read("charlie1.txt", "hello alice & bob"),
+		),
+		as(bob,
+			read("alice.txt", "hello bob & charlie"),
+			read("bob1.txt", "hello alice & charlie"),
+			read("bob2.txt", "hello alice & charlie"),
+			read("charlie1.txt", "hello alice & bob"),
+		),
+
+		inPrivateTlf("alice,bob,charlie"),
+		as(charlie,
+			read("alice.txt", "hello bob & charlie"),
+			read("bob1.txt", "hello alice & charlie"),
+			read("bob2.txt", "hello alice & charlie"),
+			read("charlie1.txt", "hello alice & bob"),
+			mkfile("charlie2.txt", "hello alice & bob"),
+		),
+		as(alice,
+			read("alice.txt", "hello bob & charlie"),
+			read("bob1.txt", "hello alice & charlie"),
+			read("bob2.txt", "hello alice & charlie"),
+			read("charlie1.txt", "hello alice & bob"),
+			read("charlie2.txt", "hello alice & bob"),
+		),
+		as(bob,
+			read("alice.txt", "hello bob & charlie"),
+			read("bob1.txt", "hello alice & charlie"),
+			read("bob2.txt", "hello alice & charlie"),
+			read("charlie1.txt", "hello alice & bob"),
+			read("charlie2.txt", "hello alice & bob"),
+		),
+	)
+}
+
+// TestSBSConflicts tests that different folders with unresolved users
+// that get resolved to the same one don't get merged, but that one is
+// chosen and the others pick up a conflict marker.
+func TestSBSConflicts(t *testing.T) {
+	test(t,
+		users("alice", "bob", "charlie"),
+		inPrivateTlf("alice,bob,charlie@twitter"),
+		as(alice,
+			mkfile("alice1.txt", "hello bob & charlie"),
+		),
+		as(bob,
+			read("alice1.txt", "hello bob & charlie"),
+		),
+		as(charlie,
+			expectError(initRoot(), "charlie does not have read access to directory /keybase/private/alice,bob,charlie@twitter"),
+		),
+
+		inPrivateTlf("alice,bob@twitter,charlie@twitter"),
+		as(alice,
+			mkfile("alice2.txt", "hello bob & charlie"),
+		),
+		as(bob,
+			expectError(initRoot(), "bob does not have read access to directory /keybase/private/alice,bob@twitter,charlie@twitter"),
+		),
+		as(charlie,
+			expectError(initRoot(), "charlie does not have read access to directory /keybase/private/alice,bob@twitter,charlie@twitter"),
+		),
+
+		inPrivateTlf("alice,bob,charlie"),
+		as(alice,
+			mkfile("alice3.txt", "hello bob & charlie"),
+		),
+		as(bob,
+			read("alice3.txt", "hello bob & charlie"),
+		),
+		as(charlie,
+			read("alice3.txt", "hello bob & charlie"),
+		),
+
+		addNewAssertion("bob", "bob@twitter"),
+		addNewAssertion("charlie", "charlie@twitter"),
+		as(alice,
+			// TODO: Ideally, we wouldn't have to do this,
+			// and we'd just wait for a rekey.
+			rekey(),
+		),
+
+		// TODO: Test that alice's favorites are updated.
+
+		// TODO: Test that the three folders are resolved with
+		// conflict markers. Need some way to control how the
+		// conflict markers are created, first...
```

Comment:
I don't think `MDServerLocal` does nothing with conflicts right now, so that part will take a bit of effort.  I don't think you need to control which conflict markers are created, it should be determined by the order you do `addNewAssertion`s.